src/ConnectedMode/Binding/ImportBeforeFileGenerator.cs

```diff
@@ -0,0 +1,107 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.IO;
+using System.IO.Abstractions;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Integration;
+
+namespace SonarLint.VisualStudio.ConnectedMode.Binding
+{
+    internal class ImportBeforeFileGenerator
+    {
+        private readonly IFileSystem fileSystem;
+        private readonly ILogger logger;
+
+        private const string targetsFileName = "SonarLint.targets";
+
+        public ImportBeforeFileGenerator(ILogger logger) : this(logger, new FileSystem()) { }
+
+        public ImportBeforeFileGenerator(ILogger logger, IFileSystem fileSystem)
+        {
+            this.logger = logger;
+            this.fileSystem = fileSystem;
+
+            WriteTargetsFileToDiskIfNotExists();
+        }
+
+        private void WriteTargetsFileToDiskIfNotExists()
+        {
+            logger.LogVerbose(Resources.ImportBeforeFileGenerator_CheckingIfFileExists);
+
+            var resource = ReadResourceFile();
```

Comment:
Minor naming suggestions: I'd probably name the variable and the method differently e.g.

```
  var fileContent = GetTargetFileContent();
```

Calling the variable `fileContent` rather than `resource` makes it clearer later in the method what the variable contains
e.g.
```
  fileSystem.File.WriteAllText(fullPath, fileContent);

  // vs

  fileSystem.File.WriteAllText(fullPath, resource);
```


Calling the method `ReadResourceFile` tells the reader where the data is coming from, but not what it represents.